### ISSUE TITLE
Make assumptions about indexed symbols

### ISSUE THREAD

**Issue by normalhuman** [sympy/sympy#13992]
I think this should be possible to do: 
```
>>> x = IndexedBase("x", positive=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/sympy/sympy/tensor/indexed.py", line 403, in __new__
    obj = Expr.__new__(cls, label, **kw_args)
TypeError: __new__() got an unexpected keyword argument 'positive'
```

Without assumptions, it's hard to do algebra with indexed symbols, e.g., get `sqrt(x[1]**2)` to simplify to `x[1]`. 

I understand that the class Indexed inherits from Expr, not from Symbol, and there are no assumptions on Expr. Still, we already have some quasi-assumptions on Indexed and IndexedBase in the form of `is_commutative=True` class property.  And the objects of class Idx, which also inherits from Expr, have them in the form `obj._assumptions["finite"] = True` as [seen here](https://github.com/sympy/sympy/blob/master/sympy/tensor/indexed.py#L604).

Could `IndexedBase.__new__ ` parse and use the `**kwargs` that it accepts, instead of trying to pass them to `Expr.__new__`? 

Optionally, maybe even `Indexed.__new__` could do that, so an assumption can be added to a particular indexed object?  

By the way, any attempt to pass `**kwargs` to `Expr.__new__` is a potential bug because this method, inherited from `Basic.__new__`, does not accept any keyword arguments.

**Comment by RavicharanN** [comment]
I would like to work on this 👍

**Comment by Nirvan101** [comment]
Can I work on this?

**Comment by normalhuman** [comment]
Yes, you can.

**Comment by jlumpe** [comment]
@Nirvan101 any progress on this?

**Comment by PymZoR** [comment]
+1

**Comment by asmeurer** [comment]
Since it doesn't subclass from Symbol, it should reapply the logic in the Symbol constructor: https://github.com/sympy/sympy/blob/cd98ba006b5c6d6a6d072eafa28ea6d0ebdaf0e7/sympy/core/symbol.py#L233-L235

Specifically, create a `StdFactKB` from the assumptions that are passed in and set `obj._assumptions` to it.

**Comment by bsamseth** [comment]
As this had been open for a while, I've made a PR that I believe resolves this issue (#16085). I based the changes on @asmeurer's comment, and have reused the logic applied in ``Symbol``.
